uaclient/contract.py

```diff
@@ -171,15 +169,24 @@ def request_machine_token_refresh(
             machine_id = util.get_machine_id(self.cfg.data_dir)
         headers = self.headers()
         headers.update({"Authorization": "Bearer {}".format(machine_token)})
-        url = API_V1_TMPL_CONTEXT_MACHINE_TOKEN_REFRESH.format(
+        url = API_V1_TMPL_CONTEXT_MACHINE_TOKEN_UPDATE.format(
             contract=contract_id, machine=machine_id
         )
-        response, headers = self.request_url(url, headers=headers)
+        data = self._get_platform_data(machine_id)
+        response, headers = self.request_url(url, headers=headers, data=data)
         if headers.get("expires"):
             response["expires"] = headers["expires"]
         self.cfg.write_cache("machine-token", response)
         return response
 
+    def _get_platform_data(self, machine_id):
+        """"Return a dict of platform-relateddata for contract requests"""
+        if not machine_id:
```

Comment:
This logic exists in both of the callsites, we don't need to replicate it in all three functions.

Reply:
fixed.